src/Combobox/index.tsx

```diff
@@ -115,19 +156,23 @@ const Combobox = ({
 }) => {
   const allChildren = React.Children.toArray(children);
```

Comment:
the type errors below are because children might not all be ReactElements - they could be string, or undefined, or other things which would cause errors.  you could tighten the props type to restrict just element-type nodes, or you could filter this `allChildren` using a [type predicate](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) function

```ts
const hasDisplayName = (e: any): e is React.ReactElement =>
  !!e?.type?.displayName;

const allChildren = React.Children.toArray(children).filter(hasDisplayName);
```

then allChildren would be compatible with the filtering below

Reply:
Wow, this is pretty cool